src/wallet/wallet.cpp

```diff
@@ -1678,8 +1682,14 @@ std::optional<ZTXOSelector> CWallet::ZTXOSelectorForAddress(const libzcash::Paym
                 // determine a local account.
                 auto accountId = this->GetUnifiedAccountId(ufvkId.value());
                 if (accountId.has_value()) {
-                    pattern = AccountZTXOPattern(accountId.value(), ua.GetKnownReceiverTypes());
+                    if (allowAddressLinkability) {
+                        pattern = AccountZTXOPattern(accountId.value(), ua.GetKnownReceiverTypes());
+                    } else {
+                        pattern = ua;
+                    }
                 }
+            } else {
+                pattern = ua;
```

Comment:
Hmm, I'm not sure if we want this. If we don't have a UFVK for the UA then we won't ever have any shielded notes for it. We _might_ have transparent coins, but only if the t-addr receiver has been separately imported - in which case that receiver would count as a legacy account.

Reply:
My thought was that we don't want to handle that problem at this layer - we're making the check as to whether or not we have the spending key elsewhere, so I think this is the right behavior for just constructing the selector.

Reply:
Hmm, okay. I guess the distinction here is that when we do have a UFVK for the UA, we at least ensure we have a corresponding account.